Drop custom toJS() / toJSNewlyCreated() functions for CSSStyleValue

cdumez opened the pull request:
#### 27ec4caed72d3e84130caf00c84de2b7a3df5566
<pre>
Drop custom toJS() / toJSNewlyCreated() functions for CSSStyleValue
<a href="https://bugs.webkit.org/show_bug.cgi?id=300229">https://bugs.webkit.org/show_bug.cgi?id=300229</a>

Reviewed by Darin Adler.

Drop custom toJS() / toJSNewlyCreated() functions for CSSStyleValue
now that it is no longer needed after 301038@main.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/JSCSSStyleValueCustom.cpp: Removed.
* Source/WebCore/bindings/js/JSCSSTransformComponentCustom.cpp:
(WebCore::toJSNewlyCreated):
* Source/WebCore/css/typedom/CSSKeywordValue.h:
(isType):
* Source/WebCore/css/typedom/CSSNumericValue.h:
(isType):
* Source/WebCore/css/typedom/CSSStyleImageValue.h:
(isType):
* Source/WebCore/css/typedom/CSSStyleValue.h:
(WebCore::isCSSNumericValue):
(WebCore::isCSSMathValue):
(WebCore::isCSSColorValue):
(WebCore::CSSStyleValue::styleValueType const):
(WebCore::CSSStyleValue::getType const): Deleted.
* Source/WebCore/css/typedom/CSSStyleValue.idl:
* Source/WebCore/css/typedom/CSSUnitValue.h:
(isType):
* Source/WebCore/css/typedom/CSSUnparsedValue.h:
(isType):
* Source/WebCore/css/typedom/color/CSSHSL.h:
(isType):
* Source/WebCore/css/typedom/color/CSSHWB.h:
(isType):
* Source/WebCore/css/typedom/color/CSSLCH.h:
(isType):
* Source/WebCore/css/typedom/color/CSSLab.h:
(isType):
* Source/WebCore/css/typedom/color/CSSOKLCH.h:
(isType):
* Source/WebCore/css/typedom/color/CSSOKLab.h:
(isType):
* Source/WebCore/css/typedom/color/CSSOMColor.h:
(isType):
(WebCore::CSSOMColor::create): Deleted.
(WebCore::CSSOMColor::alpha const): Deleted.
(WebCore::CSSOMColor::setAlpha): Deleted.
* Source/WebCore/css/typedom/color/CSSOMColorValue.h:
(isType):
(WebCore::CSSOMColorValue::isCSSHSL const): Deleted.
(WebCore::CSSOMColorValue::isCSSHWB const): Deleted.
(WebCore::CSSOMColorValue::isCSSOKLCH const): Deleted.
(WebCore::CSSOMColorValue::isCSSLCH const): Deleted.
(WebCore::CSSOMColorValue::isCSSLab const): Deleted.
(WebCore::CSSOMColorValue::isCSSOKLab const): Deleted.
(WebCore::CSSOMColorValue::isCSSOMColor const): Deleted.
(WebCore::CSSOMColorValue::isCSSRGB const): Deleted.
* Source/WebCore/css/typedom/color/CSSRGB.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathClamp.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathInvert.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathMax.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathMin.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathNegate.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathProduct.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathSum.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathValue.h:
(WebCore::CSSMathValue::equalsImpl const):
(isType):
* Source/WebCore/css/typedom/transform/CSSMatrixComponent.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSPerspective.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSRotate.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSScale.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSSkew.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSSkewX.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSSkewY.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSTransformComponent.h:
* Source/WebCore/css/typedom/transform/CSSTransformValue.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSTranslate.h:
(isType):

Canonical link: <a href="https://commits.webkit.org/301109@main">https://commits.webkit.org/301109@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8406d2434ac912add66593efb9360b5c2ce560cd

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124961 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44630 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35367 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131811 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76842 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c9a40e8d-5a59-488c-aa55-17faf3afb380) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126838 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45323 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53197 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95107 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63103 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/107d7fcb-0cf7-409d-98ca-ac1f30fc4278) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127915 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36167 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111759 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75655 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d0d6927f-d20c-4ea0-b464-ff19942c293a) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/35098 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29912 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75291 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105930 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30143 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134484 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51788 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39588 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103586 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52211 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107972 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103362 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26316 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48704 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26991 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48812 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51669 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57468 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51047 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54403 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52740 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

darinadler reviewed (approved):
Surprised to find it's named `getType` and not `type`.

cdumez committed:
Drop custom toJS() / toJSNewlyCreated() functions for CSSStyleValue
https://bugs.webkit.org/show_bug.cgi?id=300229

Reviewed by Darin Adler.

Drop custom toJS() / toJSNewlyCreated() functions for CSSStyleValue
now that it is no longer needed after 301038@main.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/JSCSSStyleValueCustom.cpp: Removed.
* Source/WebCore/bindings/js/JSCSSTransformComponentCustom.cpp:
(WebCore::toJSNewlyCreated):
* Source/WebCore/css/typedom/CSSKeywordValue.h:
(isType):
* Source/WebCore/css/typedom/CSSNumericValue.h:
(isType):
* Source/WebCore/css/typedom/CSSStyleImageValue.h:
(isType):
* Source/WebCore/css/typedom/CSSStyleValue.h:
(WebCore::isCSSNumericValue):
(WebCore::isCSSMathValue):
(WebCore::isCSSColorValue):
(WebCore::CSSStyleValue::styleValueType const):
(WebCore::CSSStyleValue::getType const): Deleted.
* Source/WebCore/css/typedom/CSSStyleValue.idl:
* Source/WebCore/css/typedom/CSSUnitValue.h:
(isType):
* Source/WebCore/css/typedom/CSSUnparsedValue.h:
(isType):
* Source/WebCore/css/typedom/color/CSSHSL.h:
(isType):
* Source/WebCore/css/typedom/color/CSSHWB.h:
(isType):
* Source/WebCore/css/typedom/color/CSSLCH.h:
(isType):
* Source/WebCore/css/typedom/color/CSSLab.h:
(isType):
* Source/WebCore/css/typedom/color/CSSOKLCH.h:
(isType):
* Source/WebCore/css/typedom/color/CSSOKLab.h:
(isType):
* Source/WebCore/css/typedom/color/CSSOMColor.h:
(isType):
(WebCore::CSSOMColor::create): Deleted.
(WebCore::CSSOMColor::alpha const): Deleted.
(WebCore::CSSOMColor::setAlpha): Deleted.
* Source/WebCore/css/typedom/color/CSSOMColorValue.h:
(isType):
(WebCore::CSSOMColorValue::isCSSHSL const): Deleted.
(WebCore::CSSOMColorValue::isCSSHWB const): Deleted.
(WebCore::CSSOMColorValue::isCSSOKLCH const): Deleted.
(WebCore::CSSOMColorValue::isCSSLCH const): Deleted.
(WebCore::CSSOMColorValue::isCSSLab const): Deleted.
(WebCore::CSSOMColorValue::isCSSOKLab const): Deleted.
(WebCore::CSSOMColorValue::isCSSOMColor const): Deleted.
(WebCore::CSSOMColorValue::isCSSRGB const): Deleted.
* Source/WebCore/css/typedom/color/CSSRGB.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathClamp.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathInvert.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathMax.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathMin.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathNegate.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathProduct.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathSum.h:
(isType):
* Source/WebCore/css/typedom/numeric/CSSMathValue.h:
(WebCore::CSSMathValue::equalsImpl const):
(isType):
* Source/WebCore/css/typedom/transform/CSSMatrixComponent.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSPerspective.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSRotate.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSScale.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSSkew.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSSkewX.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSSkewY.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSTransformComponent.h:
* Source/WebCore/css/typedom/transform/CSSTransformValue.h:
(isType):
* Source/WebCore/css/typedom/transform/CSSTranslate.h:
(isType):

Canonical link: https://commits.webkit.org/301109@main
